src/sentry/api/endpoints/group_tagkey_values.py

```diff
@@ -34,16 +35,10 @@ def get(self, request, group, key):
         """
         lookup_key = tagstore.prefix_reserved_key(key)
 
-        try:
-            environment_id = self._get_environment_id_from_request(
-                request, group.project.organization_id
-            )
-        except Environment.DoesNotExist:
-            # if the environment doesn't exist then the tag can't possibly exist
-            raise ResourceDoesNotExist
+        environment_ids = [e.id for e in get_environments(request, group.project.organization)]
 
         try:
-            tagstore.get_tag_key(group.project_id, environment_id, lookup_key)
+            tagstore.get_tag_key(group.project_id, None, lookup_key)
```

Comment:
This is probably fine, but does change behaviour a little. Previously, we'd raise `ResourceDoesNotExist` here if the tag didn't exist within the environment, now we'll only raise if it doesn't exist at all. With your other changes I think this is ok though.